src/cascadia/TerminalApp/AppLogic.cpp

```diff
@@ -905,14 +885,15 @@ namespace winrt::TerminalApp::implementation
             // editors, who will write a temp file, then rename it to be the
             // actual file you wrote. So listen for that too.
             wil::FolderChangeEvents::FileName | wil::FolderChangeEvents::LastWriteTime,
-            [this, settingsBasename = settingsPath.filename(), stateBasename = statePath.filename()](wil::FolderChangeEvent, PCWSTR fileModified) {
-                const auto modifiedBasename = std::filesystem::path{ fileModified }.filename();
+            [this, settingsBasename = settingsPath.filename()](wil::FolderChangeEvent, PCWSTR fileModified) {
+                static const auto appState{ ApplicationState::SharedInstance() };

```

Comment:
🤔

Reply:
YEP THIS WAS IT. This static reference to the app state was keeping it alive after the `ActionToStringMap` was cleaned up, so when we try to persist the actions in the window state, we couldn't. We'd try to look up the action and the map just, wouldn't fuckin exist, and it'd explode, and the Terminal wouldn't know what to do when it explodes there, and the Terminal was tearing down anyways, so it'd just die. Of course, this isn't good enough for Windbg, who'd tell that something was awry, and would try to post-mortem attach to the (exiting) Terminal. This would fail, and then windbg would give you the ["i tried my best"](https://media3.giphy.com/media/tJibdDBnGmFr2/200.gif) warning I've been seeing all morning.